extensions/amp-story/1.0/amp-story-system-layer.js

```diff
@@ -97,86 +97,79 @@ const renderSystemLayerElement = (element, children) => (
-        )}
+        data-aria-label={LocalizedStringId_Enum.AMP_STORY_INFO_BUTTON_LABEL}
       />
       <div class="i-amphtml-story-sound-display">
         <button
           class={UNMUTE_CLASS + ' i-amphtml-story-button'}
-          aria-label={localize(
-            element,
+          data-aria-label={
             LocalizedStringId_Enum.AMP_STORY_AUDIO_UNMUTE_BUTTON_LABEL
-          )}
+          }
         />
         <button
           class={MUTE_CLASS + ' i-amphtml-story-button'}
-          aria-label={localize(
-            element,
+          data-aria-label={
             LocalizedStringId_Enum.AMP_STORY_AUDIO_MUTE_BUTTON_LABEL
-          )}
+          }
         />
       </div>
       <div class="i-amphtml-paused-display">
         <button
           class={PAUSE_CLASS + ' i-amphtml-story-button'}
-          aria-label={localize(
-            element,
-            LocalizedStringId_Enum.AMP_STORY_PAUSE_BUTTON_LABEL
-          )}
+          data-aria-label={LocalizedStringId_Enum.AMP_STORY_PAUSE_BUTTON_LABEL}
         />
         <button
           class={PLAY_CLASS + ' i-amphtml-story-button'}
-          aria-label={localize(
-            element,
-            LocalizedStringId_Enum.AMP_STORY_PLAY_BUTTON_LABEL
-          )}
+          data-aria-label={LocalizedStringId_Enum.AMP_STORY_PLAY_BUTTON_LABEL}
         />
       </div>
       <button
         class={
           SKIP_TO_NEXT_CLASS +
           ' i-amphtml-story-ui-hide-button i-amphtml-story-button'
         }
-        aria-label={localize(
-          element,
+        data-aria-label={
           LocalizedStringId_Enum.AMP_STORY_SKIP_TO_NEXT_BUTTON_LABEL
-        )}
+        }
       />
       <button
         class={SHARE_CLASS + ' i-amphtml-story-button'}
-        aria-label={localize(
-          element,
-          LocalizedStringId_Enum.AMP_STORY_SHARE_BUTTON_LABEL
-        )}
+        data-aria-label={LocalizedStringId_Enum.AMP_STORY_SHARE_BUTTON_LABEL}
       />
       <button
         class={
           CLOSE_CLASS + ' i-amphtml-story-ui-hide-button i-amphtml-story-button'
         }
-        aria-label={localize(
-          element,
-          LocalizedStringId_Enum.AMP_STORY_CLOSE_BUTTON_LABEL
-        )}
+        data-aria-label={LocalizedStringId_Enum.AMP_STORY_CLOSE_BUTTON_LABEL}
       />
     </div>
     <div class="i-amphtml-story-system-layer-buttons-start-position" />
   </aside>
 );
 
+const localizeSystemLayer = (systemLayer, parentElement) => {
+  const localizationService = getLocalizationService(parentElement);
+  systemLayer.querySelectorAll('[data-aria-label]').forEach((el) => {
+    localizationService.localizeEl(
+      el,
+      el.getAttribute('data-aria-label'),
+      'aria-label'
+    );
+    el.removeAttribute('data-aria-label');
+  });
+  localizationService.localizeEl(
+    systemLayer.querySelector('.i-amphtml-story-has-new-page-text'),
+    LocalizedStringId_Enum.AMP_STORY_HAS_NEW_PAGE_TEXT
+  );
+};
+
```

Comment:
We shouldn't have any LocalizedStringId_Enum key in here, this should just be a post processing function using the data provided by developers within templates.

Also, this should be a helper that can be re-used for all other templates in other files.

A few additional questions:

1. How will you block rendering on strings that need to be rendered synchronously
2. If you add a string asynchronously but it is visible to users, then it needs to run in a mutateElement context

Reply:
1. Block rendering can happen like so: `localize(template).then(() => {document.appendChild(template)})`
2. Yes, in that scenario it will likely be a render-blocking localization, which currently happens in a mutate context so we just do that after the promise.